tests/dataloaders/test_dataloaders.py

```diff
@@ -92,29 +95,74 @@ def test_anndataloader_distributed_sampler_init():
 
 
 def multiprocessing_worker(
-    rank: int, world_size: int, manager: scvi.data.AnnDataManager, save_path: str
+    rank: int,
+    world_size: int,
+    manager: scvi.data.AnnDataManager,
+    save_path: str,
+    datasplitter_kwargs,
 ):
     # initializes the distributed backend that takes care of synchronizing processes
     torch.distributed.init_process_group(
-        "gloo",  # backend that works on all systems
+        "nccl",  # backend that works on all systems
         init_method=f"file://{save_path}/dist_file",
         rank=rank,
         world_size=world_size,
+        store=None,
     )
 
-    _ = scvi.dataloaders.AnnDataLoader(manager, distributed_sampler=True)
+    _ = scvi.dataloaders.AnnDataLoader(manager, **datasplitter_kwargs)
 
     return
 
 
-@pytest.mark.optional
-def test_anndataloader_distributed_sampler(save_path: str, num_processes: int = 2):
-    adata = scvi.data.synthetic_iid()
-    manager = generic_setup_adata_manager(adata)
+@pytest.mark.parametrize("num_processes", [1, 2])
+def test_anndataloader_distributed_sampler(num_processes: int, save_path: str):
+    if torch.cuda.is_available():
```

Comment:
Shouldn't it be assert here?

Reply:
So, there is no such thing as a CUDA flag and I dont want the test to fail on a CPU run.